lib/secure_headers/headers/content_security_policy.rb

```diff
@@ -108,15 +109,32 @@ def build_media_type_list_directive(directive)
     def build_source_list_directive(directive)
       source_list = @config.directive_value(directive)
       if source_list != OPT_OUT && source_list && source_list.any?
-        minified_source_list = minify_source_list(directive, source_list).join(" ")
+        cleaned_source_list = clean_malformatted_sources(directive, source_list)
+        minified_source_list = minify_source_list(directive, cleaned_source_list).join(" ")
+        [symbol_to_hyphen_case(directive), minified_source_list].join(" ").strip
+      end
+    end
 
-        if minified_source_list =~ /(\n|;)/
-          Kernel.warn("#{directive} contains a #{$1} in #{minified_source_list.inspect} which will raise an error in future versions. It has been replaced with a blank space.")
+    # Private: Calculates a modified version version of source_list:
+    # - where any expression containing a deprecated character (semicolon or newline) has been split by that
+    # character, and
+    # - any empty string expression is removed.
+    def clean_malformatted_sources(directive, source_list)
+      cleaned_source_list = []
+      semicolon_warned_yet = false
+      source_list.map do |expression|
+        if expression =~ /(\n|;)/
+          if !semicolon_warned_yet
+            Kernel.warn("#{directive} contains a #{$1} in #{source_list.join(" ").inspect} which will raise an error in future versions. It has been replaced with a blank space.")
```

Comment:
Likely the `$1` in the warning here will be very confusing if it’s an actual new line. Could we do `$1.inspect`? I think that will show `”\n”`